### PR TITLE
feat: add animation on deploy button in cd modal

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
add animation on deploy button in cd modal

Fixes https://github.com/devtron-labs/sprint-tasks/issues/1574

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/material/MaterialList.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ import { CreateMaterial } from './CreateMaterial'
 import { UpdateMaterial } from './UpdateMaterial'
 import { MaterialListProps, MaterialListState } from './material.types'
 import { ReactComponent as GitHub } from '../../assets/icons/ic-sample-app.svg'
-import { ReactComponent as PlayMedia } from '../../assets/icons/ic-play-media.svg'
+import { ReactComponent as PlayMedia } from '../../assets/icons/ic-play-outline.svg'
```
Comment (AbhishekA1509) [inline src/components/material/MaterialList.tsx:32]: Please use alias

File: src/components/app/details/triggerView/cdMaterial.tsx [hunk 2]
```diff
@@ -1653,30 +1659,33 @@ const CDMaterial = ({
...
-                <Progressing />
-            ) : (
-                <>
-                    {getDeployButtonIcon()}
-                    {deploymentWindowMetadata.userActionState === ACTION_STATE.BLOCKED
+    const renderTriggerDeployButton = (disableDeployButton: boolean) => {
+        if (
+            stageType === DeploymentNodeType.CD &&
+            !disableDeployButton &&
+            deploymentWindowMetadata?.userActionState === ACTION_STATE.ALLOWED &&
```
Comment (AbhishekA1509) [inline src/components/app/details/triggerView/cdMaterial.tsx:1666]: This won't work if deploymentWindowMetadata is null

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: yarn.lock [hunk 3]
```diff
@@ -160,7 +160,7 @@
   dependencies:
     "@babel/types" "^7.24.7"
 
-"@babel/helper-plugin-utils@^7.0.0", "@babel/helper-plugin-utils@^7.10.4", "@babel/helper-plugin-utils@^7.12.13", "@babel/helper-plugin-utils@^7.14.5", "@babel/helper-plugin-utils@^7.18.6", "@babel/helper-plugin-utils@^7.22.5", "@babel/helper-plugin-utils@^7.24.7", "@babel/helper-plugin-utils@^7.24.8", "@babel/helper-plugin-utils@^7.8.0", "@babel/helper-plugin-utils@^7.8.3":
+"@babel/helper-plugin-utils@^7.0.0", "@babel/helper-plugin-utils@^7.10.4", "@babel/helper-plugin-utils@^7.12.13", "@babel/helper-plugin-utils@^7.14.5", "@babel/helper-plugin-utils@^7.18.6", "@babel/helper-plugin-utils@^7.22.5", "@babel/helper-plugin-utils@^7.24.7", "@babel/helper-plugin-utils@^7.24.8", "@babel/helper-plugin-utils@^7.25.9", "@babel/helper-plugin-utils@^7.8.0", "@babel/helper-plugin-utils@^7.8.3":
```
Comment (AbhishekA1509) [inline yarn.lock:163]: Is it intentional

Comment (arunjaindev) [reply yarn.lock]: reverted

File: src/components/app/details/triggerView/cdMaterial.tsx [hunk 4]
```diff
@@ -1653,30 +1659,32 @@ const CDMaterial = ({
...
+                startIcon={getDeployButtonIcon()}
+                isLoading={deploymentLoading || isSaveLoading}
+                text={`${
+                    userActionState === ACTION_STATE.BLOCKED ? 'Deployment is blocked' : CDButtonLabelMap[stageType]
+                }${isVirtualEnvironment ? ' to isolated env' : ''}`}
+                endIcon={userActionState === ACTION_STATE.BLOCKED ? <InfoOutline /> : null}
+                onClick={(e) => onClickDeploy(e, disableDeployButton)}
+                size={ComponentSizeType.large}
+                style={getDeployButtonStyle(userActionState)}
+                disabled={disableDeployButton}
```
Comment (AbhishekA1509) [inline src/components/app/details/triggerView/cdMaterial.tsx:1684]: Will it be disabled in case of deploymentLoading || isSaveLoading

Comment (arunjaindev) [reply src/components/app/details/triggerView/cdMaterial.tsx:1684]: <Button /> Component handles disabling on isLoading i.e. deploymentLoading || isSaveLoading in this case

File: src/assets/icons/misc/arrow-solid-right.svg
Comment (vivek-devtron) [inline src/assets/icons/misc/arrow-solid-right.svg:1]: Have we verified the impact?

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**